test/puppetlabs/puppetdb/testutils.clj

```diff
@@ -440,3 +441,37 @@
         table]
        jdbc/query-to-vec
        (map :indexdef)))
+
+(defn update-timestamp [resource resource-list time]
```

Comment:
Correct me if I'm confused here, but this looks like it's updating the timestamps of events in the fixture data based on their index in the `resources` vector. Then we're using the fact that we're getting results ordered by this timestamp back from the db so that the test fixture order matches the query result. I think this is happening due to the `sort-by` found in [testutils/reports/munge-resource-events](https://github.com/puppetlabs/puppetdb/blob/6.x/test/puppetlabs/puppetdb/testutils/reports.clj#L149) but not totally sure yet. We might want to take some time at the meeting tomorrow to double check this and see if we might be able to carry over the difference between the existing timestamps in the fixture data when we do the update.